src/sentry/api_gateway/proxy.py

```diff
@@ -45,6 +45,7 @@ def proxy_request(request: Request, org_slug: str) -> StreamingHttpResponse:
     request_args = {
         "headers": request.headers,
         "params": dict(query_params) if query_params is not None else None,
+        "data": getattr(request, "body", None),
```

Comment:
Do, we know that this also does not buffer?  Like I know we stream the response body, but does this properly stream the request body, too?  It might, it's just not clear, curious what the docs say,.

Reply:
So if the body is JSON, I think we always load it to memory in the endpoint. if it's a file stream object, then the request.body will have a `file` field which contains a buffer for the file contents. 

You can see that here: https://github.com/getsentry/sentry/blob/47f7aa7789416a0b3feec2d9517850156aeb42bd/src/sentry/api/endpoints/debug_files.py#L50

Reply:
Got it.  So it's not a completely pass through proxy if we're buffering on disk.

I know I'm being super pedantic here, but it's on my mind: is there any way we can "trick" the django endpoint _not to buffer or consume the body using the default logic_ for this case?

Like, ideally, we'd have a fixed size memory buffer per request that does buffer _some_ of the request until it is consumed by the proxied endpoint, but it is valuable that the fixed size memory buffer be reasonably small so as to prevent larger endpoints from clogging control silo's memory or disk.  This is also important because doing deserialization and reserialization of request bodies is non trivial cpu expense.  This problem grows continuously with each new region AND single tenant customer (something that sales is actively looking to increase).

But I'm game to hear other thoughts?  @getsentry/hybrid-cloud

Reply:
While I think it would be good to have fixed size memory buffers eventually, I don't think we need it in this first iteration. This seems like a good topic for backend TSC.

Reply:
Sure.  I won't block the PR and I agree that we need a first iteration, but I don't think we should let this fall off the radar.  Thanks for the back and forth.